### ISSUE TITLE
Silent breakage on gfx 0.6.5

### ISSUE THREAD

**Issue by bvssvni** [gfx-rs/gfx#786]
I just fixed https://github.com/PistonDevelopers/piston_window/issues/62 where I suddenly got this error:

```
/Users/sven/.cargo/registry/src/github.com-0a35038f75765ae4/piston_window-0.8.0/src/lib.rs:33:62: 33:90 error: wrong number of type arguments: expected 1, found 0 [E0243]
/Users/sven/.cargo/registry/src/github.com-0a35038f75765ae4/piston_window-0.8.0/src/lib.rs:33 pub type G2d<'a> = GfxGraphics<'a, gfx_device_gl::Resources, gfx_device_gl::CommandBuffer, gfx_device_gl::Output>;
                                                                                                                                                           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/sven/.cargo/registry/src/github.com-0a35038f75765ae4/piston_window-0.8.0/src/lib.rs:33:62: 33:90 help: run `rustc --explain E0243` to see a detailed explanation
/Users/sven/.cargo/registry/src/github.com-0a35038f75765ae4/piston_window-0.8.0/src/lib.rs:130:39: 130:67 error: wrong number of type arguments: expected 1, found 0 [E0243]
/Users/sven/.cargo/registry/src/github.com-0a35038f75765ae4/piston_window-0.8.0/src/lib.rs:130             gfx_device_gl::Resources, gfx_device_gl::CommandBuffer,
                                                                                                                                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/sven/.cargo/registry/src/github.com-0a35038f75765ae4/piston_window-0.8.0/src/lib.rs:130:39: 130:67 help: run `rustc --explain E0243` to see a detailed explanation
error: aborting due to 2 previous errors
Could not compile `piston_window`.
```

This was committed on https://github.com/gfx-rs/gfx/commit/180d4a15e2cb5ba345e7b6b24f4b548e6a7ea100#diff-b42bd9be6c2e5cf60482e1230ad3007c but the first non-zero version was not increased.

I believe this happened by accident, see https://github.com/gfx-rs/gfx/issues/736.

I've also written up more extensively on the motivation to increase the first non-zero number for breaking changes https://github.com/PistonDevelopers/piston/issues/931.

**Comment by kvark** [comment]
Same as with Rust versions, there is always a compromise. If a theoretically breaking change (which applies to most changes) doesn't break anything in practice, we increase only the patch version. Unfortunately, this one did break `piston_window`, please accept our apologies!

**Comment by csherratt** [comment]
We can yank `0.6.5` and make it into `0.7.0`.

**Comment by bvssvni** [comment]
I agree there is a compromise, but don't think it applies in this situation. Judging whether something breaks anything in practice can be difficult and I don't expect this be done perfectly. As a gfx contributor and getting notifications I should be on guard for things that might be breaking changes, so I am partly responsible. Let's see if we can avoid more breaking changes until 0.7. We have this problem everywhere, so if it turns out this is a big problem then perhaps we should figure out some way to auto detect breaking changes and report it.

**Comment by kvark** [comment]
Thanks @bvssvni, :+1: for the plan!

**Comment by bvssvni** [comment]
Notice that the post about incrementing for breaking changes predates this issue https://github.com/PistonDevelopers/piston/issues/931. I added a section about minimizing damage based on the things I learned, though. If there any ideas of how to improve it I am open to suggestions. If this remains a problem I might write a blog post about it to get the idea out there.

I'll close this since the issue is solved now.
